test/test_torch.py

```diff
@@ -12303,6 +12309,24 @@ def non_zero_rand(size, dtype, device):
                     non_zero_rand((2, 2), dtype=dtype, device=device),
                     non_zero_rand((2, 2), dtype=dtype, device=device))
 
+    # This function tests that a nan value is returned for input values not in domain
+    def test_acosh_domain(self, device):
+        # Domain of acosh is [1, inf)
+        sample = torch.tensor([-1.23, -0.06, 0.98], device=device)
+        self.assertEqual(torch.isnan(torch.acosh(sample)), torch.BoolTensor([1, 1, 1]))
```

Comment:
torch.BoolTensor is outdated. Instead do this before the comparison:

```
nan_mask = torch.tensor((True, True, True, ..., False), device=device)
```
```

Reply:
Sounds good, this is fixed. Thanks, @mruberry!